test/verify_reflective_test_suites.dart

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/dart/analysis/session.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/file_system/file_system.dart';
+import 'package:analyzer/file_system/physical_file_system.dart';
+import 'package:path/path.dart' as path;
+import 'package:test/test.dart';
+
+void main() {
+  group('check reflective test suites', () {
+    group('rules', () {
+      var testDirPath = PhysicalResourceProvider.INSTANCE.pathContext
+          .absolute('test', 'rules');
+      _VerifyTests(testDirPath).build();
+    });
+  });
+}
+
+/// Helper class to test that [testAllFileName] files are properly set up.
+/// (Cribbed from `analyzer_utilities`.)
+class _VerifyTests {
```

Comment:
I will guess that this is a copy of what we have in SDK, so I will not review it carefully.

Reply:
Yeah. I trimmed it a bit since our needs are simpler.